bin/reth/src/commands/node/mod.rs

```diff
@@ -128,6 +132,77 @@ pub struct NodeCommand<Ext: RethCliExt = ()> {
 }
 
 impl<Ext: RethCliExt> NodeCommand<Ext> {
+    /// Launches the node
+    ///
+    /// This transforms the node command into a node config and launches the node using the given
+    /// closure.
+    pub async fn execute2<L, Fut>(self, ctx: CliContext, launcher: L) -> eyre::Result<()>
+    where
+        L: FnOnce(WithLaunchContext<Arc<DatabaseEnv>, InitState>, Ext::Node) -> Fut,
+        Fut: Future<Output = eyre::Result<()>>,
+    {
+        let Self {
+            datadir,
+            config,
+            chain,
+            metrics,
+            trusted_setup_file,
+            instance,
+            with_unused_ports,
+            network,
+            rpc,
+            txpool,
+            builder,
+            debug,
+            db,
+            dev,
+            pruning,
+            #[cfg(feature = "optimism")]
+            rollup,
+            ext,
+        } = self;
```

Comment:
dont love this long destructure do we want a `From<NodeCommand> for NodeConfig` impl?